### PR TITLE
Fix useless asserts in installer tests

### PR DESCRIPTION
**Pull request by rmynar**
Fixing assertions that were always True and therefore useless.

Note: `bool("any string with non-zero length")` is True, `True or False` is True

### CODE REVIEW

**Comment by rmynar** [comment]
trigger: test-robottelo
pytest: tests/foreman/ -k 'test_installer_sat_pub_directory_accessibility or test_installer_cap_pub_directory_accessibility'

**Comment by ogajduse** [comment]
Good catch @rmynar! I have prepared a patch that adds flake8-simplify and flake8-return rulesets to our ruff linter config some time ago and seeing your patch made me publish it today. #14100 

Let's merge yours first and I will rebase mine.

**Review by ogajduse — APPROVED** [review]

**Comment by ogajduse** [comment]
@rmynar Can you please take a look at the PRT failures?

**Comment by rmynar** [comment]
trigger: test-robottelo
pytest: tests/foreman/ -k "test_installer_sat_pub_directory_accessibility or test_installer_cap_pub_directory_accessibility"

File: tests/foreman/destructive/test_installer.py [hunk 1]
```diff
@@ -63,8 +63,8 @@ def test_installer_sat_pub_directory_accessibility(target_sat):
     http_curl_command = f'curl -i {target_sat.url.replace("https", "http")}/pub/ -k'
     https_curl_command = f'curl -i {target_sat.url}/pub/ -k'
     for command in [http_curl_command, https_curl_command]:
-        accessibility_check = target_sat.execute(command)
-        assert 'HTTP/1.1 200 OK' or 'HTTP/2 200 ' in accessibility_check.stdout.split('\r\n')
+        accessibility_check = target_sat.execute(command).stdout.split('\r\n')
+        assert 'HTTP/1.1 200 OK' in accessibility_check or 'HTTP/2 200 ' in accessibility_check
```
Comment (JacobCallahan) [inline tests/foreman/destructive/test_installer.py:67]: If you're just checking for specific substrings, you likely don't need to bother splitting the output into a list.
```suggestion
        accessibility_check = target_sat.execute(command).stdout
        assert 'HTTP/1.1 200 OK' in accessibility_check or 'HTTP/2 200 ' in accessibility_check
```

File: tests/foreman/destructive/test_installer.py [hunk 2]
```diff
@@ -73,8 +73,9 @@ def test_installer_sat_pub_directory_accessibility(target_sat):
     command_output = target_sat.execute('satellite-installer', timeout='20m')
     assert 'Success!' in command_output.stdout
     for command in [http_curl_command, https_curl_command]:
-        accessibility_check = target_sat.execute(command)
-        assert 'HTTP/1.1 200 OK' or 'HTTP/2 200 ' not in accessibility_check.stdout.split('\r\n')
+        accessibility_check = target_sat.execute(command).stdout.split('\r\n')
+        assert 'HTTP/1.1 200 OK' not in accessibility_check
+        assert 'HTTP/2 200 ' not in accessibility_check
```
Comment (JacobCallahan) [inline tests/foreman/destructive/test_installer.py:78]: Same recommendation here
```suggestion
        accessibility_check = target_sat.execute(command).stdout
        assert 'HTTP/1.1 200 OK' not in accessibility_check
        assert 'HTTP/2 200 ' not in accessibility_check
```

File: tests/foreman/installer/test_installer.py [hunk 3]
```diff
@@ -1755,8 +1755,8 @@ def test_installer_cap_pub_directory_accessibility(capsule_configured):
     http_curl_command = f'curl -i {capsule_configured.url.replace("https", "http")}/pub/ -k'
     https_curl_command = f'curl -i {capsule_configured.url}/pub/ -k'
     for command in [http_curl_command, https_curl_command]:
-        accessibility_check = capsule_configured.execute(command)
-        assert 'HTTP/1.1 200 OK' or 'HTTP/2 200 ' in accessibility_check.stdout.split('\r\n')
+        accessibility_check = capsule_configured.execute(command).stdout.split('\r\n')
+        assert 'HTTP/1.1 200 OK' in accessibility_check or 'HTTP/2 200 ' in accessibility_check
```
Comment (JacobCallahan) [inline tests/foreman/installer/test_installer.py:1759]: You get the idea
```suggestion
        accessibility_check = capsule_configured.execute(command).stdout
        assert 'HTTP/1.1 200 OK' in accessibility_check or 'HTTP/2 200 ' in accessibility_check
```

File: tests/foreman/installer/test_installer.py [hunk 4]
```diff
@@ -1765,8 +1765,9 @@ def test_installer_cap_pub_directory_accessibility(capsule_configured):
     command_output = capsule_configured.execute('satellite-installer', timeout='20m')
     assert 'Success!' in command_output.stdout
     for command in [http_curl_command, https_curl_command]:
-        accessibility_check = capsule_configured.execute(command)
-        assert 'HTTP/1.1 200 OK' or 'HTTP/2 200 ' not in accessibility_check.stdout.split('\r\n')
+        accessibility_check = capsule_configured.execute(command).stdout.split('\r\n')
+        assert 'HTTP/1.1 200 OK' not in accessibility_check
+        assert 'HTTP/2 200 ' not in accessibility_check
```
Comment (JacobCallahan) [inline tests/foreman/installer/test_installer.py:1770]: Same
```suggestion
        accessibility_check = capsule_configured.execute(command).stdout
        assert 'HTTP/1.1 200 OK' not in accessibility_check
        assert 'HTTP/2 200 ' not in accessibility_check
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Some minor recommendations

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_installer.py::test_installer_sat_pub_directory_accessibility tests/foreman/installer/test_installer.py::test_installer_cap_pub_directory_accessibility

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5780
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/destructive/test_installer.py::test_installer_sat_pub_directory_accessibility tests/foreman/installer/test_installer.py::test_installer_cap_pub_directory_accessibility
Test Result : ================= 2 passed, 114 warnings in 2647.34s (0:44:07) =================
```

**Review by lhellebr — APPROVED** [review]
I agree with Jacob, otherwise LGTM

**Comment by rmynar** [comment]
I just wanted to fix the logical error in assertion, but you're right that the split was useless. Hopefully it looks better now

**Pull request merged**